lib/services/marketplace-plugins-service.ts

```diff
@@ -12,8 +12,8 @@ export class MarketplacePluginsService implements ICordovaPluginsService {
 		return this.$server.cordova.getMarketplacePluginsData(this.$project.projectData.Framework);
 	}
 
-	public createPluginData(plugin: IMarketplacePluginVersionsData): IMarketplacePlugin[] { // DefaultVersion, Identifier, Versions
-		return _.map(plugin.Versions, (pluginVersionData) => new PluginsDataLib.MarketplacePluginData(plugin, pluginVersionData, this.$project, this.$projectConstants));
+	public createPluginData(plugin: IMarketplacePluginVersionsDataBase): IMarketplacePlugin[] { // DefaultVersion, Identifier, Versions
+		return <any[]>_.map(plugin.Versions, (pluginVersionData) => new PluginsDataLib.MarketplacePluginData(<any>plugin, <any>pluginVersionData, this.$project, this.$projectConstants));
```

Comment:
If you change the method's return type (as suggested by the interface in fact) then no casting is required.
Case in point: 
```Typescript
public createPluginData(plugin: IMarketplacePluginVersionsDataBase): IPlugin[] { // DefaultVersion, Identifier, Versions
	return _.map(plugin.Versions, (pluginVersionData) => new PluginsDataLib.MarketplacePluginData(plugin, pluginVersionData, this.$project, this.$projectConstants));
}
```